compiler/crates/relay-lsp/src/rename/mod.rs

```diff
@@ -0,0 +1,616 @@
+        finder.argument_locations
+    }
+
+    fn add_argument_location(&mut self, name_location: common::Location) {
+        let location_without_dollar = name_location.with_span(Span {
+            start: name_location.span().start + 1,
+            end: name_location.span().end,
+        });
+
+        self.argument_locations.push(location_without_dollar);
+    }
+}
+
+impl Visitor for FragmentArgumentFinder {
+    const NAME: &'static str = "FragmentArgumentFinder";
+    const VISIT_ARGUMENTS: bool = true;
+    const VISIT_DIRECTIVES: bool = true;
+
+    fn visit_fragment_spread(&mut self, spread: &FragmentSpread) {
+        if spread.fragment.item.0 == self.fragment_name {
+            spread
+                .arguments
+                .iter()
+                .filter(|a| a.name.item.0 == self.argument_name)
+                .for_each(|a| {
+                    self.argument_locations.push(a.name.location);
+                })
+        }
+    }
+
+    fn visit_argument(&mut self, argument: &graphql_ir::Argument) {
+        if let Some(fragment_name) = self.current_scope.fragment_name {
+            if fragment_name == self.fragment_name {
+                match &argument.value.item {
+                    graphql_ir::Value::Variable(variable) => {
+                        if variable.name.item.0 == self.argument_name {
+                            self.add_argument_location(variable.name.location);
+                        }
+                    }
+                    _ => (),
+                }
+            }
+        }
+    }
+
+    fn visit_fragment(&mut self, fragment: &FragmentDefinition) {
+        assert!(self.current_scope.fragment_name.is_none());
+        self.current_scope.fragment_name = Some(fragment.name.item.0);
+
+        if fragment.name.item.0 == self.fragment_name {
+            let argument_name = self.argument_name;
+            let has_argument_definitions_directive = fragment
+                .directives
+                .named(DirectiveName(*ARGUMENTDEFINITIONS_DIRECTIVE))
+                .is_some();
+
+            fragment
+                .variable_definitions
+                .iter()
+                .filter(|v| v.name.item.0 == argument_name)
+                .for_each(|v| {
+                    if has_argument_definitions_directive {
+                        // If the argument was defined in @argumentDefinitions it doesn't start
+                        // with a dollar sign, so we add the full name without transformations.
+                        self.argument_locations.push(v.name.location);
+                    } else {
+                        self.add_argument_location(v.name.location)
+                    }
+                });
+        }
+
+        self.default_visit_fragment(fragment);
+
+        self.current_scope.fragment_name = None;
+    }
+
+    // For some reason the argument on @skip isn't visited,
+    // but the "argument variable" is visited through this...
```

Comment:
I think conditionals are transformed in the IR into their own node type. So @skip and @include get transformed into their or IR node

Reply:
Good to know. Would you recommend a different approach that what I've taken here?

Reply:
Removed :)

Reply:
I think this is fine.